[ESIMD][DOC][NFC] Add invoke_simd() section to README; update ESIMD e… (#9145)
intel/llvm: sycl/doc/extensions/experimental/sycl_ext_intel_esimd/sycl_ext_intel_esimd.md

## Introduction

The main motivation for introducing the "Explicit SIMD" SYCL extension 

> (or simply "ESIMD") is enabling low-level efficient programming for Intel graphics

Reviewer: ```suggestion
(or simply "ESIMD") is enabling efficient low-level programming for Intel graphics
```

> architectures. It provides APIs closely matching Intel GPU ISA

Reviewer: ```suggestion
architectures. It provides APIs close to the Intel GPU ISA
```

> yet allows to write explicitly vectorized device code.

Reviewer: ```suggestion
and allows writing explicitly vectorized device code.
```

Explicit SIMD provides the following key features complementary to SYCL:
- Manual vectorization of device code using the `simd` class mapped to Intel GPU's
  general register file. This allows to write efficient code not relying on
  further widening by the compiler, as with traditional SPMD programming.

> - Low-level APIs efficiently mapped to Intel GPU architecture, such as block loads/stores/gathers/scatters, explicit cache hints, GPU inline assembly, etc.

Reviewer: ```suggestion
- Low-level APIs efficiently mapped to the Intel GPU architecture, such as block loads/stores/gathers/scatters, explicit cache hints, GPU inline assembly, etc.
```

## Implementation restrictions

Current ESIMD implementation does not support using certain standard SYCL features
inside explicit SIMD kernels and functions. Most of them will be eventually
dropped. What's not supported today:
- Explicit SIMD kernels can co-exist with regular SYCL kernels in the same
  translation unit and in the same program.

> - Interoperability between regular SYCL and ESIMD kernels is supported one way only.

Reviewer: ```suggestion
- Interoperability between regular SYCL and ESIMD kernels is only supported one way.
```
